lib/carto/visualization_exporter.rb

```diff
@@ -9,20 +9,65 @@ def initialize(http_client = Carto::Http::Client.get('data_exporter', log_reques
     end
 
     # Returns the file
-    def export_table(user_table, folder, format)
+    def export_table(user_table, folder, format, is_remote)
       table_name = user_table.name
 
       query = %{select * from "#{table_name}"}
       url = sql_api_query_url(query, table_name, user_table.user, privacy(user_table), format)
       exported_file = "#{folder}/#{table_name}.#{format}"
+
       @http_client.get_file(url, exported_file, ssl_verifypeer: false, ssl_verifyhost: 0)
+
+      # Hack for Samples 2.0 - Save As
+      if(format == 'gpkg')
+        # Verify if the data is remote or if it is local
+        if(is_remote)
+          Carto::GpkgCartoMetadataUtil.open( geopkg_file: exported_file ) do |md|
+            md.metadata = { 
+              vendor: 'carto',
+              data: {
+                source: {
+                  type: 'fdw',
+                  configuration: {
+                    parent_table: table_name
+                  }
+                }
+              }
+            }.with_indifferent_access
+            
+            # Clear the data
+            md.clear_table(table_name: table_name)
+          end
```

Comment:
Is the else part taken care somewhere else? (for the standard cartodb export)

Reply:
The standard export is taken care of as part of sql_api_query_url().  This if statement "intercepts" the standard cartodb export file and updates it as appropriate.  Therefore there is no else.